chapters/chap03-sec03.tex

```diff
@@ -114,45 +117,46 @@ \subsection{Instantiating the packing functions}\label{instantiation:statement:p
-            \t r \gets r \concat \slice{\htags{i}}{\floor{\prfPkOutLen / \bnFieldBitCap} \cdot \bnFieldBitCap}{}; \\
+            \t r \gets \slice{\nfs{i}}{\floor{\prfNfOutLen / \bnFieldBitCap} \cdot \bnFieldBitCap}{} \concat r; \\
+            %
+            r \gets \slice{\hsig}{\floor{\crhhsigOutLen / \bnFieldBitCap} \cdot \bnFieldBitCap}{} \concat r; \\
             %
             \pcfor i \in [ \ceil{\len{r}/ \bnFieldBitCap} ] \pcdo: \\
             \t out \gets \pack{\slice{r}{i \cdot \bnFieldBitCap}{ (i+1) \cdot \bnFieldBitCap}}{\FFx{\rBN}}; \\
             \pcreturn out;
         }
-        \caption{Packing residual bits algorithm.}\label{zeth-protocol:fig:packing-resbits-alg}
+        \caption{Algorithm to pack residual bits.}\label{zeth-protocol:fig:packing-resbits-alg}
     \end{minipage}
 \end{figure*}
 
 \subsubsection{Packing Policy Security}
 \begin{proposition}[Packing security]
-    For a variable $x$ it holds $\unpack{\pack{x}{}}{} = x$ and 
-    $\unpack{\packResBits{x}}{} = x$.
+    For a binary string $x$, it holds that $\unpack{\pack{x}{}}{} = x$ and $\unpack{\packResBits{x}}{} = x$.
 \end{proposition}
 
 \subsubsection{Packing Policy Example}
 
-In the case where $\jsin=\jsout=2$, $\bnFieldBitCap$ and all PRFs, and $\crhhsig{}$ output 256 bits, the unpacked primary inputs are 2167-bit long. The packing parameters thus are,
+In the case where $\jsin=\jsout=2$, field elements hold $\bnFieldBitCap$ bits and all PRFs and $\crhhsig{}$ output bit-strings of length 256, the unpacked primary inputs are 2167-bit long.
+The packing parameters are therefore:
 \begin{align*}
     & \resBitsBLen = 143 \\
     & \nfFLen = \hsigFLen = \htagFLen = \resBitsFLen = 1
 \end{align*}
-The packed primary inputs are 2277 bits long which corresponds to a small space overhead ($\approx 5\%$ unused bits). Moreover, as the residual bits are 143-bit long, they can be written over a single field element. As such, the primary inputs are 9-field element long.
+The packed primary inputs are 2277 bits long, corresponding to a small space overhead of $\approx 5\%$ unused bits. Moreover, the 143-bit residual bits can be packed into a single field element. As such, the primary inputs are encoded as 9 field elements.
 Finally, the residual bits are formatted as follows,
 \[
-    \underbrace{padding}_{\text{113 bits}} \concat \underbrace{\vin\vphantom{g}}_{64\ bits} \concat \underbrace{\vout\vphantom{g}}_{64\ bits} \concat \underbrace{\hsig}_{3\ bits} \concat \underbrace{\nf{0}}_{3\ bits} \concat \underbrace{\nf{1}}_{3\ bits} \concat \underbrace{\htag{0}}_{3\ bits} \concat \underbrace{\htag{1}}_{3\ bits}
+    \underbrace{padding}_{\text{113 bits}} \concat \underbrace{\hsig}_{3\ bits} \concat \underbrace{\nf{0}}_{3\ bits} \concat \underbrace{\nf{1}}_{3\ bits} \concat \underbrace{\htag{0}}_{3\ bits} \concat \underbrace{\htag{1}}_{3\ bits} \concat \underbrace{\vin\vphantom{g}}_{64\ bits} \concat \underbrace{\vout\vphantom{g}}_{64\ bits}
```

Comment:
Hum.. based on the algo above, shouldn't it be: `hsig || nf1 || nf0 || htag1 || htag0...`?

Reply:
Fixed this.  Thanks.